hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/metrics/ContractMetrics.java

```diff
@@ -174,48 +313,143 @@ public void bumpRejectedType3EthTx(final long bumpBy) {
+            @NonNull SystemContractMethod systemContractMethod, @NonNull final MessageFrame.State state) {
+        systemContractMethod =
+                systemContractMethodRegistry.fromMissingContractGetWithContract(requireNonNull(systemContractMethod));
+        requireNonNull(state);
+
+        final var result = MethodResult.from(state);
+        if (p2MetricsEnabled) {
+            // Handle each of the P2 metrics kinds
+
+            try {
+                final Consumer<Counter[]> bumpMetricsPair = metricsPair -> {
+                    metricsPair[MethodMetricType.TOTAL.index].increment();
+                    if (MethodResult.FAILURE == result) {
+                        metricsPair[MethodMetricType.FAILED.index].increment();
+                    }
+                };
+
+                bumpMetricsPair.accept(systemContractMethodCounters.get(
+                        systemContractMethod.systemContract().orElse(null)));
+                bumpMetricsPair.accept(systemContractMethodCountersVia.get(systemContractMethod.via()));
+                systemContractMethodErcMembers
+                        .get(systemContractMethod)
+                        .forEach(ercType -> bumpMetricsPair.accept(systemContractERCTypeCounters.get(ercType)));
+                systemContractMethodGroupMembers
+                        .get(systemContractMethod)
+                        .forEach(group -> bumpMetricsPair.accept(systemContractMethodGroupCounters.get(group)));
+            } catch (final NullPointerException e) {
+                // ignore: should never happen, but ... just in case ... don't fail tx because of bad
+                // metrics code
+            }
+        }
+    }
+
+    private final ConcurrentHashMap<SystemContractMethod, SystemContractMethod> methodsThatHaveCallsWithNullMethod =
+            new ConcurrentHashMap<>(50);
+
+    public void logWarnMissingSystemContractMethodOnCall(@NonNull final SystemContractMethod systemContractMethod) {
+        // Log, but only first time you see it for a specific method (to avoid spew)
+        requireNonNull(systemContractMethod);
+        if (methodsThatHaveCallsWithNullMethod.containsKey(systemContractMethod)) {
+            log.warn("Found `Call` without `SystemContractMethod`,  %s"
+                    .formatted(systemContractMethod.fullyDecoratedMethodName()));
+        } else {
+            methodsThatHaveCallsWithNullMethod.put(systemContractMethod, systemContractMethod);
+        }
+    }
+
+    // -----------------
+    // Unit test helpers
+
+    @VisibleForTesting
+    public @NonNull Set<Counter> getAllP1Counters() {
+        final var allCounters = new HashSet<Counter>(200);
+        allCounters.addAll(rejectedTxsCounters.values());
+        allCounters.addAll(rejectedTxsLackingIntrinsicGas.values());
+        if (rejectedEthType3Counter != null) allCounters.add(rejectedEthType3Counter);
+        return allCounters;
+    }
+
+    @VisibleForTesting
```

Comment:
I didn't see any unit tests that required this annotation - here and many of the others.  Are they coming later?